src/osha/oira/statistics/utils.py

```diff
@@ -159,10 +188,18 @@ def update_assessment(self, country=None):
 
         def assessment_rows(offset):
             batch = sessions.limit(self.b_size).offset(offset)
-            rows = [
-                SurveySessionStatistics(
+            handled = 0
+            for session, account, total_risks, answered_risks in batch:
+                existing = (
+                    self.session_statistics.query(SurveySessionStatistics)
+                    .filter(SurveySessionStatistics.id == session.id)
+                    .first()
+                )
```

Comment:
If the same session is present more than one time in the batch you will get redundant queries

Reply:
`id` is the primary key, so if I get multiple results for this I would say that I have much bigger problems than redundant queries, or am I missing something?

Reply:
I have no idea what that batch returns ;)
That's why I said *if* ;)

If you say that the batch will not return something like:
```
[
    (sessionA, ...),
    (sessionA, ...)
    ...
    (sessionB, ...)
]
```
I am fine with that.

Actually, I am fine with that anyway but I was just pointing the attention on that because the point of the PR is optimization and speed improvement :)

Reply:
Ah, now I see what you mean. It won't, but thanks for pointing it out!